keanu-project/src/main/java/io/improbable/keanu/tensor/generic/GenericTensor.java

```diff
@@ -63,13 +65,13 @@ public boolean isShapePlaceholder() {
     }
 
     @Override
-    public T getValue(int... index) {
-        return data[getFlatIndex(shape, stride, index)];
+    public T getValue(long... index) {
+        return data[Ints.saturatedCast(getFlatIndex(shape, stride, index))];
```

Comment:
We should do a check here and throw if it's above or below max or min instead of saturated casting.

Reply:
see my comment above - use `Math.toIntExact`